### PR TITLE
Elixir + Phoenix + Node.js + PostgreSQL Dev Container revival

### PR DESCRIPTION
**Pull request by talk2MeGooseman**
Hey, this PR is a revival of PR #432. I noticed that it had gone stale after September and would really love to have this Elixir + Phoenix + Node + Postgres dev container finished and merged.

I have addressed the comments in the PR (#432) and added one or two more nice to have in the configurations. If you have any more feedback just let me know. I love using Dev Containers (and Github Codespaces) and excited to contribute to something awesome.

### CODE REVIEW

File: containers/elixir-phoenix-postgres/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Update the VARIANT arg in devcontainer.json to pick an Elixir version: 1.9, 1.10, 1.10.4
```
Comment (Chuxel) [inline containers/elixir-phoenix-postgres/.devcontainer/Dockerfile:1]: Comment tweak: needs to be updated in docker-compose.yml.

File: containers/elixir-phoenix-postgres/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	],
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	"forwardPorts": [4000, 4001, 5432],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "mix deps.get"
+
+	// Uncomment to connect as a non-root user. See https://aka.ms/vscode-remote/containers/non-root.
+	// "remoteUser": "vscode"
```
Comment (Chuxel) [inline containers/elixir-phoenix-postgres/.devcontainer/devcontainer.json:36]: Can you update this to default to `vscode`? We switched our standard recently.

File: containers/elixir-phoenix-postgres/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+### Adding the definition to your folder
+
+1. If this is your first time using a development container, please follow the [getting started steps](https://aka.ms/vscode-remote/containers/getting-started) to set up your machine.
+
+2. To use VS Code's copy of this definition:
+
+   1. Start VS Code and open your project folder.
+   2. Press <kbd>F1</kbd> select and **Remote-Containers: Add Development Container Configuration Files...** from the command palette.
+   3. Select the Ruby definition.
```
Comment (Chuxel) [inline containers/elixir-phoenix-postgres/README.md:55]: Wrong definition name in README here.

**Review by Chuxel — COMMENTED** [review]
This is looking great!

One thing we're now doing is defaulting to the "vscode" user rather than root. However, when I switched to vscode, running `test.sh` failed likely due to something missing from the PATH or environment.

```bash
$ mix phx.new example --live

** (Mix) The task "phx.new" could not be found
Note no mix.exs was found in the current directory
```

But this works as root

```bash
$ sudo mix phx.new example --live 

* creating example/config/config.exs
* creating example/config/dev.exs
* creating example/config/prod.exs
* creating example/config/prod.secret.exs
...
```
Otherwise this is looking good!

**Comment by talk2MeGooseman** [comment]
@Chuxel Thanks for the review and testing out the code. Found the issue with trying to run things as the non-root user. The user doesn't have access to anything installed by the root user with mix so you have to switch to that user before then in the Dockerfile.
I am not 100% great with the Docker-foo so let me know if this is a valid solution.

**Comment by Chuxel** [comment]
@talk2MeGooseman Yep, that works, but to avoid the extra layer and leave the image running as root by default, you can use the `su` command to temporarily run as the other user.  e.g.

```bash
RUN su ${USERNAME} -c "mix local.hex --force \
  && mix local.rebar --force \
  && mix archive.install --force hex phx_new ${PHOENIX_VERSION}"
 ```

**Comment by talk2MeGooseman** [comment]
@Chuxel Thanks for the feedback and awesome tip

**Comment by talk2MeGooseman** [comment]
@PavelSosin-320 The image capable of running both on a local machine and within Codespaces. Curious to know if you see any glaring red flags? So I can keep a lookout for those.

**Comment by Chuxel** [comment]
LGTM!  Thanks!

**Comment by PavelSosin-320** [comment]
@talk2MeGooseman The Docker's "Checker flag" is the correct configuration and implementation of the HEALTHCHECK [Docker healthcheck](https://docs.docker.com/engine/reference/run/#healthcheck).
If I can configure docker run --healthcheck options in such a way that ensures that no Docker default timeout, no network timeout no ctrl/c from the user will happen because of the slow container instantiation of the Image it is OK for me regardless of how many things it contains.  Otherwise, I suspect that the user will have trouble with it. According to the Murphy law, it will certainly happen. Since docker ps will not contain any useful information about container status it will be "it is always OK on my computer" issue.

**Pull request merged**